modules/replicator/src/main/java/org/apache/ignite/internal/raft/client/TopologyAwareRaftGroupService.java

```diff
@@ -514,5 +539,55 @@ void resetLeader() {
     @Override
     public void updateConfiguration(PeersAndLearners configuration) {
         this.raftClient.updateConfiguration(configuration);
+
+        List<CompletableFuture<?>> futures = new ArrayList<>();
+
+        for (Peer peer : peers()) {
+            // Subscribe to all new peers in configuration.
+            if (!subscribers.containsKey(peer)) {
```

Comment:
Could there be a race `containsKey` 1 and `put`?

Reply:
These operations shouldn't happen concurrently, but I'll make the code more safe in case they could in the future